wrappers/java/src/main/java/org/hyperledger/indy/sdk/agent/Agent.java

```diff
@@ -0,0 +1,329 @@
+package org.hyperledger.indy.sdk.agent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+import org.hyperledger.indy.sdk.LibSovrin;
+import org.hyperledger.indy.sdk.SovrinException;
+import org.hyperledger.indy.sdk.SovrinJava;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentAddIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseConnectionResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseListenerResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentConnectResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentListenResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentRemoveIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentSendResult;
+import org.hyperledger.indy.sdk.pool.Pool;
+import org.hyperledger.indy.sdk.wallet.Wallet;
+
+import com.sun.jna.Callback;
+
+/**
+ * agent.rs API
+ */
+public class Agent extends SovrinJava.API {
+
+	private Agent() {
+
+	}
+
+	/*
+	 * STATIC METHODS
+	 */
+
+	public static Future<AgentConnectResult> agentConnect(
+			Pool pool,
+			Wallet wallet,
+			String senderDid,
+			String receiverDid,
+			Callback messageCb) throws SovrinException {
+
+		final CompletableFuture<AgentConnectResult> future = new CompletableFuture<> ();
+
+		Callback connectionCb = new Callback() {
+
+			@SuppressWarnings("unused")
+			public void callback(int xcommand_handle, int err, int connection_handle) {
+
+				if (! checkCallback(future, xcommand_handle, err)) return;
+
+				Agent.Connection connection = new Agent.Connection(connection_handle);
+
+				AgentConnectResult result = new AgentConnectResult(connection);
+				future.complete(result);
+			}
+		};
+
+		int poolHandle = pool.getPoolHandle();
+		int walletHandle = wallet.getWalletHandle();
+
+		int result = LibSovrin.api.sovrin_agent_connect(
+				FIXED_COMMAND_HANDLE, 
+				poolHandle,
+				walletHandle, 
+				senderDid,
+				receiverDid,
+				connectionCb,
+				messageCb);
+
+		checkResult(result);
+
+		return future;
+	}
+
+	public static Future<AgentListenResult> agentListen(
```

Comment:
> Callback connectionCb
> Callback messageCb

These callbacks are low-level c callbacks that use low-level ids, handles and error codes as params. It would be nice to hide this details from library user. I suggest using observer-like approach. We can define one interface 

```
interface ConnectionObserver {
  void onConnection(AgentConnection connection);
  void onMessage(AgentConnection connection, String message);
}
```

And ask a user to pass implementation of this observer as the latest parameter of agentConnect. This parameter will replace both callbacks.